packages/core-ethereum/src/ethereum.ts

```diff
@@ -193,9 +174,48 @@ export async function createChainWrapper(
       throw new Error(`Failed in mining transaction. ${error}`)
     }
 
-    // Give other tasks CPU time to happen
-    // Allow the indexer to update before awaiting the response
-    await setImmediate()
+    log('Transaction with nonce %d successfully sent %s, waiting for confimation', nonce, transaction.hash)
+    transactions.moveFromQueuingToPending(transaction.hash)
+    nonceLock.releaseLock()
+
+    try {
+      // wait for the tx to be mined - mininal and scheduled implementation
+      // only fails if tx does not get mined within the specified timeout
+      await new Promise<void>((resolve, reject) => {
+        let done = false
+        const cleanUp = (err?: string) => {
+          if (done) {
+            return
+          }
+          done = true
+
+          provider.off(transaction.hash, onTransaction)
+          if (err) {
+            setImmediate(reject, Error(err))
```

Comment:
What is the benefit of pausing here? Please document.

Reply:
changed